roman-shuvalov/nodejs-aws-be
# task9: add bff

product-service: 
https://zuejyaeeb7.execute-api.eu-west-1.amazonaws.com/dev
bff-product-service:
http://roman-shuvalov-bff-api-dev.eu-west-1.elasticbeanstalk.com/products

cart-service:
http://roman-shuvalov-cart-api-dev.eu-west-1.elasticbeanstalk.com/api
bff-cart-service:
http://roman-shuvalov-bff-api-dev.eu-west-1.elasticbeanstalk.com/profile/cart

Example of create new product request:
```
curl -d '{"count":1,"description":"P4","price":1,"title":"P4"}' \
  -H "Content-Type: application/json" \
  -X POST http://roman-shuvalov-bff-api-dev.eu-west-1.elasticbeanstalk.com/products
```

Evaluation criteria
**3 (done)** - A working and correct express application should be in the bff-service folder. Reviewer can start this application locally with any valid configuration in the .env file and this application should works as described in the task 9.1
**5 (done)** - The bff-service should be deployed with Elastic Beanstalk. The bff-service call should be redirected to the appropriate service : product-service or CART. The response from the bff-service should be the same as if product-service or CART services were called directly.

Additional (optional) tasks
**+1 (done)** - Add a cache at the bff-service level for a request to the getProductsList function of the product-service. The cache should expire in 2 minutes.
How to test:
- Get products list
- Create new product
- Get products list - result shouldn’t have new product
- Wait more than 2 minutes
- Get products list - result should have new product

**+1 (done)** - Use NestJS to create bff-service instead of express
